kubespawner/spawner.py

```diff
@@ -1809,18 +1836,32 @@ def _expand_user_properties(self, template):
         safe_username = escapism.escape(
             self.user.name, safe=safe_chars, escape_char='-'
         ).lower()
+
+        if self.slug_scheme == "safe":
+            # safe slug scheme escapes _after_ rendering the template
+            username = self.user.name
+            servername = raw_servername
+        else:
+            username = safe_username
+            servername = safe_servername
+
         rendered = template.format(
             userid=self.user.id,
-            username=safe_username,
+            username=username,
+            escaped_username=safe_username,
```

Comment:
Do we really need this extra parameter ?

Reply:
to ensure the same fields can still be used in the new scheme, yes if anyone needs to make sure names are preserved across schemes